<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
    <meta name="description" content="">
    <meta name="author" content="">
    
    <script async src="https://www.googletagmanager.com/gtag/js?id=UA-174131763-1"></script>
    <script>
      window.dataLayer = window.dataLayer || [];
      function gtag(){dataLayer.push(arguments);}
      gtag('js', new Date());
      gtag('config', 'UA-174131763-1');
    </script>
    
    <script defer src="https://use.fontawesome.com/releases/v5.0.8/js/fontawesome.js"
     integrity="sha384-7ox8Q2yzO/uWircfojVuCQOZl+ZZBg2D2J5nkpLqzH1HY0C1dHlTKIbpRz/LG23c" crossorigin="anonymous"></script>

    <title>Basic site for interactive tSNE plotting</title>

    <!-- Bootstrap core CSS -->
    <link href="css/bootstrap.min.css" rel="stylesheet">

    <!-- Custom styles for this template -->
    <link href="css/style.css" rel="stylesheet">

    <!-- css -->
    <!-- <link rel="stylesheet" type="text/css" href="www/site_stylesheet.css"/> -->
    <!-- Import Vega & Vega-Lite (does not have to be from CDN) -->
    <script src="https://cdn.jsdelivr.net/npm/vega@5"></script>
    <script src="https://cdn.jsdelivr.net/npm/vega-lite@4"></script>
    <!-- Import vega-embed -->
    <script src="https://cdn.jsdelivr.net/npm/vega-embed@6"></script>

  </head>
  <body>
    <a name="top"></a>
    <main role="main" class="container">
      <div class="container">
       <h1>Interactive 2D tSNE plotting of cell-specific methylation and gene expression markers</h1>
        <p>
         This page provides an interactive companion to the data that is detailed in our recent publication [DOI: <a href="https://dx.doi.org/10.21203/rs.2.13274/v1" target="blank">10.21203/rs.2.13274/v1</a>]. <br>
         Code and data for all plots on this page can be found <a href="https://github.com/sirselim/tSNE_plotting" target="blank">here</a>. Data, figures and additional files supporting our publication can be found <a href="https://github.com/sirselim/immunecell_methylation_paper_data" target="blank">here</a>.
        </p>
        <figure>
        <p><i><b>Note:</b> The plots below are fully interactive and are able to be dragged and zoomed using the mouse. To reset the plot double click anywhere on it. The legends are also interactive, to select a group click on it with the mouse. To make multiple selections hold down 'shift' and click each group with the mouse. Again, double clicking anywhere on the plot will reset to the default view.</i></p>
        <div id="vis" style="text-align:center"></div>
        <figcaption><b>Figure:</b> tSNE plot of all 1173 methylation CpG sites which were selected as being cell type markers.</figcaption>
        </figure>
        <p style="width: 500px;">
          <blockquote><b>Citation:</b>
            <i>Donia Macartney-Coxson, Alanna Cameron, Jane Clapham, and Miles C Benton. DNA methylation in blood - potential to provide new insights in immune cell biology, 
            20 August 2019, PREPRINT (Version 1) available at Research Square [DOI: <a href="https://dx.doi.org/10.21203/rs.2.13274/v1" target="blank">10.21203/rs.2.13274/v1</a>]</i></blockquote>
        </p>
        <br>
        <hr>
        <br>
        <h4>Exploring cell-specific markers in other public methylation array data sets</h4>
        We set out to determine if the markers identified above would perform in independent public cell-sorted data sets. We identified the below studies as appropriate candidates for this validation:
        <ul>
          <li><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE103541" target="blank">GSE103541</a> - [Illumina EPIC] 145 samples (<b>140 used</b> in this validation); <b>DNA methylation profiles for CD4 T cells, CD8 T cells, B cells, Monocytes and Granulocytes purified from 28 individuals.</b></li>
          <li><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE110554" target="blank">GSE110554</a> - [Illumina EPIC] 49 samples (<b>49 used</b> in this validation); Salas LA, Koestler DC, Butler RA, Hansen HM et al. <b><i>An optimized library for reference-based deconvolution of whole-blood biospecimens assayed using the Illumina Human Methylation EPIC BeadArray</i>.</b> Genome Biol 2018 May 29;19(1):64. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/29843789/" target="">29843789</a></li>
          <li><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE82084" target="blank">GSE82084</a> - [Illumina 450K] 36 samples (<b>22 used</b> in this validation); de Goede OM, Lavoie PM, Robinson WP. <b><i>Cord blood hematopoietic cells from preterm infants display altered DNA methylation patterns</i>.</b> Clin Epigenetics 2017;9:39. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/28428831/" target="blsnk">28428831</a></li>
        </ul>
        Two of these three data sets were generated on the Illumina EPIC platform, while our marker panel is derived from Illumina 450K array data. We identified a total of 1025 CpG sites that overlapped the original panel of 1173 markers. This drop is due to a combination of some probes not being present on the EPIC platform and some samples having missing data (with resultant probes being dropped).
        <br>
        <figure>
        <div id="vis2" style="text-align:center"></div>
        <figcaption><b>Figure:</b> tSNE plot of sorted-cells from 211 samples based on 1025 methylation CpG (overlapping with the three selected data sets).</figcaption>
        </figure>
        <br>
        <hr>
        The below figure is a re-creation of the above including the cell-sorted data from Reinius 2012 (<a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE35069" target="blank";>GSE35069</a>) - the data set which was used to identify the 1173 cell-specific CpG markers.
        <ul>
          <li><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE35069" target="blank";>GSE35069</a>- [Illumina 450K] 60 samples (<b>60 used</b> in this experiment); Reinius LE, Acevedo N, Joerink M, Pershagen G, Dahlén SE, et al. (2012) <b>Differential DNA Methylation in Purified Human Blood Cells: Implications for Cell Lineage and Studies on Disease Susceptibility</b>. PLOS ONE 7(7): e41361. PMID: <a href="https://doi.org/10.1371/journal.pone.0041361" target="blank">22848472</a></li>
        </ul>
        <figure>
        <div id="vis3" style="text-align:center"></div>
        <figcaption><b>Figure:</b> tSNE plot of sorted-cells from 271 samples based on 1025 methylation CpG (the three selected data sets plus the original Reinius data set used to identify the markers).</figcaption>
        </figure>
        <br>
        <hr>
        <h4>Exploring cell-specific markers in a publicly available expression data set</h4>
        The below figure depicts expression (RNASeq) data from cell-sorted populations - <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE107011" target="blank";>GSE107011</a>
        <ul>
          <li>Xu W, Monaco G, Wong EH, Tan WLW et al. (2019) <b>Mapping of γ/δ T cells reveals Vδ2+ T cells resistance to senescence.</b> EBioMedicine PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30528453" target="blank";>30528453</a></li>
        </ul>
        For this validation of our QRTPCR results we extracted data for the same five cell populations (CD4, CD8, CD19, CD56 and CD14) and then extracted expression data (TPM) for each of the 11 genes used in our study. The resultant data was analysed using t-SNE, results below:
        <figure>
        <div id="vis4" style="text-align:center"></div>
        <figcaption><b>Figure:</b> tSNE plot of sorted-cells from the RNASeq study GSE107011. This analysis was performed on expression data (TPM) from 11 genes, these are the same 11 genes that we validated the expression of in our experiment (QRTPCR)</figcaption>
        </figure>
        There are some interesting observations in the above plot:
        <ul>
          <li>the data used actually explored sub-types of specific populations:</li>
            <ul>
              <li>TE = Terminal Effector</li>
              <li>EM = Effector Memory</li>
              <li>CM = Central Memory</li>
              <li>NC_mono = Non Classical monocytes</li>
              <li>I_mono = Intermediate monocytes </li>
              <li>C_mono = Classical monocytes</li>
              <li>B_SM = Switched memory B cells</li>
              <li>B_Ex = Exhausted B cells</li>
              <li>B_NSM = Non-switched memory B cells</li>
            </ul>
          <li></li>
          <li></li>
          <li></li>
        </ul>
        <br>
        <hr>
        <h4>Testing - large visualisations</h4>
        This section explores using t-sne on a large number of methylation CpG sites and then tries to visualise the results in an interactive manner.
        <br>
        [TESTING] - this <a href="img/test_large_tsne_plot.html">link</a> explores a very large data set using <a href="https://bokeh.org/" target="blank">bokeh</a>, be warned there be dragons!
        <br>
        <hr>
        <a href="#top">Back to top of page</a>
      </div>
    </main>

    <!-- vegalite embed -->
    <script type="text/javascript">
      var spec =
        "https://raw.githubusercontent.com/sirselim/tSNE_plotting/master/scripts/tsne_scatterplot.vl.json";
      vegaEmbed("#vis", spec)
        .then(function (result) {
          // Access the Vega view instance (https://vega.github.io/vega/docs/api/view/) as result.view
        })
        .catch(console.error);
    </script>

    <!-- vegalite embed -->
    <script type="text/javascript">
      var spec =
        "https://raw.githubusercontent.com/sirselim/tSNE_plotting/master/scripts/tsne_scatterplot_public.vl.json";
      vegaEmbed("#vis2", spec)
        .then(function (result) {
          // Access the Vega view instance (https://vega.github.io/vega/docs/api/view/) as result.view
        })
        .catch(console.error);
    </script>

    <!-- vegalite embed -->
    <script type="text/javascript">
      var spec =
        "https://raw.githubusercontent.com/sirselim/tSNE_plotting/master/scripts/tsne_scatterplot_public_all.vl.json";
      vegaEmbed("#vis3", spec)
        .then(function (result) {
          // Access the Vega view instance (https://vega.github.io/vega/docs/api/view/) as result.view
        })
        .catch(console.error);
    </script>

    <!-- vegalite embed -->
    <script type="text/javascript">
      var spec =
        "https://raw.githubusercontent.com/sirselim/tSNE_plotting/master/scripts/tsne_scatterplot_exprs.vl.json";
      vegaEmbed("#vis4", spec)
        .then(function (result) {
          // Access the Vega view instance (https://vega.github.io/vega/docs/api/view/) as result.view
        })
        .catch(console.error);
    </script>

    <!-- Bootstrap core JavaScript
    ================================================== -->
    <!-- Placed at the end of the document so the pages load faster -->
    <script src="js/jquery-3.3.1.min.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.12.9/umd/popper.min.js"
    integrity="sha384-ApNbgh9B+Y1QKtv3Rn7W3mgPxhU9K/ScQsAP7hUibX39j7fakFPskvXusvfa0b4Q" crossorigin="anonymous"></script>
    <script src="js/bootstrap.min.js"></script>
  </body>
</html>
